soccer/Referee.cpp

```diff
@@ -262,167 +263,184 @@ void NewRefereeModule::spinKickWatcher() {
     }
 }
 
-void NewRefereeModule::updateGameState(bool blueTeam) {
-    _context->game_state.ourScore =
-        blueTeam ? blue_info.score : yellow_info.score;
-    _context->game_state.theirScore =
-        blueTeam ? yellow_info.score : blue_info.score;
-    using namespace NewRefereeModuleEnums;
-    switch (stage) {
-        case Stage::NORMAL_FIRST_HALF_PRE:
-            _context->game_state.period = GameState::FirstHalf;
-            break;
-        case Stage::NORMAL_FIRST_HALF:
-            _context->game_state.period = GameState::FirstHalf;
-            break;
-        case Stage::NORMAL_HALF_TIME:
-            _context->game_state.period = GameState::Halftime;
-            break;
-        case Stage::NORMAL_SECOND_HALF_PRE:
-            _context->game_state.period = GameState::SecondHalf;
-            break;
-        case Stage::NORMAL_SECOND_HALF:
-            _context->game_state.period = GameState::SecondHalf;
-            break;
-        case Stage::EXTRA_TIME_BREAK:
-            _context->game_state.period = GameState::FirstHalf;
-            break;
-        case Stage::EXTRA_FIRST_HALF_PRE:
-            _context->game_state.period = GameState::Overtime1;
-            break;
-        case Stage::EXTRA_FIRST_HALF:
-            _context->game_state.period = GameState::Overtime1;
-            break;
-        case Stage::EXTRA_HALF_TIME:
-            _context->game_state.period = GameState::Halftime;
-            break;
-        case Stage::EXTRA_SECOND_HALF_PRE:
-            _context->game_state.period = GameState::Overtime2;
-            break;
-        case Stage::EXTRA_SECOND_HALF:
-            _context->game_state.period = GameState::Overtime2;
-            break;
-        case Stage::PENALTY_SHOOTOUT_BREAK:
-            _context->game_state.period = GameState::PenaltyShootout;
-            break;
-        case Stage::PENALTY_SHOOTOUT:
-            _context->game_state.period = GameState::PenaltyShootout;
-            break;
-        case Stage::POST_GAME:
-            _context->game_state.period = GameState::Overtime2;
+void Referee::update() {
+    _context->game_state = updateGameState(_context->game_state);
```

Comment:
I'll have to double check, but nothing else should update game state right? I can't find anything when I search and I think gui changes `command`.

If that's the case, I think it would fit this module structure better to have a private `game_state` variable in this class that we just keep updated and then push into context every frame. It'll mirror the pub/sub quite a bit better

Reply:
Yeah, this (and `overrideTeam` but that works with what you proposed) is the only place where it updates game state. I think it's cleaner as well, I'll make that change.